server/src/lib/model/db/migrations/20191024074951-challenge-related-schemas.ts

```diff
@@ -10,33 +10,36 @@ export const up = async function(db: any): Promise<any> {
 
       name VARCHAR(255),
       url_token VARCHAR(36) NOT NULL,
-      start_date DATETIME NOT NULL
+      start_date DATETIME NOT NULL,
+
+      UNIQUE (url_token)
     );
 
     ALTER TABLE teams DROP PRIMARY KEY,
               ADD COLUMN id INT UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY, /* better performance and shorter sql statements */
-              CHANGE invite_token url_token VARCHAR(36) NOT NULL,
+              CHANGE invite_token url_token VARCHAR(36) UNIQUE NOT NULL,
```

Comment:
This won't necessarily be unique across competitions. For instance, I think the following should be two different, but valid, teams:

- `?challenge=pilot&team=sap`
- `?challenge=other&team=sap`

Maybe we can add `UNIQUE(challenge_id, url_token)` instead?

Reply:
good point!